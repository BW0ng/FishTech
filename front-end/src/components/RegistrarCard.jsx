
import { Accordion } from "react-bootstrap";

function RegistrarCard(props) {
    return (
        <Accordion.Item eventKey="2">
            <Accordion.Header>Registrar Information</Accordion.Header>
            <Accordion.Body>
                <strong>Name: </strong>
                {props.data.registrarName}

                <br/>
                <br/>

                
                <strong>IANA ID: </strong>
                {props.data.registrarIANAID}
                <br/>
                <br/>

                <strong>Contact Phone: </strong>
                {props.data.customField2Value}
                <br/>
                <br/>

                <strong>Website: </strong>
                {props.data.customField3Value}
                
            </Accordion.Body>
        </Accordion.Item>
    );
}

export default RegistrarCard;