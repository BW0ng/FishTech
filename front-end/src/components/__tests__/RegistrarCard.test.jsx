import React from 'react';
import renderer from 'react-test-renderer';
import RegistrarCard from '../RegistrarCard.jsx';

describe('RegistrarCard', () => {
    it('renders correctly with full data output', () => {
        let input = { 
            registrarName: 'registrarName',
            registrarIANAID: 'registrarIANAID',
            customField2Value : 'customField2Value',
            customField3Value : 'customField3Value',

         };
        const component = renderer.create(<RegistrarCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });
});