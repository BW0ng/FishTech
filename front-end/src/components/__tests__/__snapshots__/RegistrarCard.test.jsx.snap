// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrarCard renders correctly with full data output 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Registrar Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      registrarName
      <br />
      <br />
      <strong>
        IANA ID: 
      </strong>
      registrarIANAID
      <br />
      <br />
      <strong>
        Contact Phone: 
      </strong>
      customField2Value
      <br />
      <br />
      <strong>
        Website: 
      </strong>
      customField3Value
    </div>
  </div>
</div>
`;
